xs2a-adapter-aspsp-registry/src/main/java/de/adorsys/xs2a/adapter/registry/exception/RegistryIOException.java

```diff
@@ -0,0 +1,11 @@
+package de.adorsys.xs2a.adapter.registry.exception;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class RegistryIOException extends RuntimeException {
+
+    public RegistryIOException(IOException cause) {
+        super(Objects.requireNonNull(cause));
```

Comment:
`Objects.requireNonNull()` throws `NPE` in case of null argument. Is it correct for us?

Reply:
Yes

Reply:
Aren't you afraid, that if NPE is thrown, `RegistryIOException` will not be thrown (as the object will not be created due to the error in the constructor), so our logs won't show the real cause of the application problem?

Reply:
How you will get the real cause if you allow for null to be passed in? The not null check ensures that that the causing IOException always associated with a RegistryIOException. Besides we shouldn't pass nulls in constructors in non-test code anyway.